type_lens/type_view.py

```diff
@@ -154,3 +154,14 @@ def has_inner_subclass_of(self, cl: type[Any] | tuple[type[Any], ...]) -> bool:
             Whether any of the type's generic args are a subclass of the given type.
         """
         return any(t.is_subclass_of(cl) for t in self.inner_types)
+
+    def strip_optional(self) -> TypeView:
```

Comment:
I know I'm a bit late to the party but I think `without_none_union` would make more sense in overall. For one, it fits the wording of the other properties better (`is_`, `has_`, etc.), and more importantly, it's closer to what's happening under the hood (same is true for `is_optional`, which probably should be something like `is_none_union`).

@peterschutt Wdyt about making these changes?

Reply:
I don't mind the fact that we use the term "optional" to mean "union with `None`", because it has that meaning assigned to it from `typing.Optional`. What about `without_optional`?

Reply:
> because it has that meaning assigned to it from `typing.Optional`. What about `without_optional`?

That *is* going away though and is being replaced by `<type> | None`, that's what I was thinking. Might be a bit of a long-term thought here but Python 3.9 *is* going to be EOL by the end of next year, so by then it's probably safe to assume `Optional` usage isn't the default anymore really and not part of the supported versions anyway.

> What about `without_optional`?

Sounds good!

Reply:
the optional verbiage isn't deprecated or going away, even if there's now syntax to avoid it.

The thing that concerns me about "without_" is that it maybe implies more than happens. `Optional[list[Optional[int]]]` would yield `list[Optional[int]]` so it's not like it's..."without" it anymore. it's doing a one-level strip of the optional that may or may not exist.

> For one, it fits the wording of the other properties better (is_, has_, etc.)
And I feel like that's not particularly meaningful personally. Every other method is a boolean check where a passive verb(?) fits more, imo.

I'm pro active `strip_` verb personally. Or perhaps something like `remove_` (which i suppose lines up with `"".removeprefix`).

Reply:
> the optional verbiage isn't deprecated or going away, even if there's now syntax to avoid it.

True, but it being less common plus it not being represented as such (i.e. `Optional` is just `| None` internally anyway), would mean that there's less of a strong connection between what an "optional" type is.

> Or perhaps something like `remove_` (which i suppose lines up with `"".removeprefix`).

I feel like `remove_` as well would imply that it removes inner `Optional`s as well.
In this case I think `strip_` is the only thing that makes clear that it's only removing the outer layer. 

However, I think `without_` communicates another aspect better: That it returns a new thing *without* this specific property. `strip_` on the other hand could also indicate that the instance is being mutated. This holds true for the stdlib as well, e.g. `list.sort()` and `dict.clear()` mutate in place. It's only on immutables that this verbiage is used to indicate the creation of a new instance, e.g. `str.strip`.

Reply:
> It's only on immutables that this verbiage is used to indicate the creation of a new instance, e.g. str.strip.

These are intended to be immutable - at least that's what I was going for by marking everything as `Final`.

Reply:
I think the overloading of the term Optional to mean union with none is unfortunate, and would be happy to see it go away. So how about `strip_none_union`?

Reply:
> I think the overloading of the term Optional to mean union with none is unfortunate, and would be happy to see it go away. So how about `strip_none_union`? And we'd rename `is_optional` to `is_none_union`.

Sounds good to me!

Reply:
i mean, i'm not the library owner, but i still think going against the verbiage of python itself, [typing_inspect](https://github.com/ilevkivskyi/typing_inspect) (maintained by a mypy maintainer seemingly), and mypy/pyright at a glance is unnecessary.

I feel like "optional" is very much a more obvious sounding name than "none_union" given the history and everything else

Reply:
I agree with @DanCardin on this. I feel like `strip_optional` makes sense over `strip_none_union.` Also, like he said while the `Optional` type might be going away, I think it's pretty common for people to mention `str | None` types as an optional string etc.

Reply:
> i still think going against the verbiage of python itself, [typing_inspect](https://github.com/ilevkivskyi/typing_inspect) (maintained by a mypy maintainer seemingly), and mypy/pyright at a glance is unnecessary.

Maybe. Sometimes it's best to go with the flow, but sometimes it's also worth going in the opposite direction. Personally, I think `Optional` to mean "a union with `None`" was a poor choice, as it creates much confusion. It might be worth considering to lead by example here how things could be improved. 

But then again, that's just my opinion. I'm not gonna block on this in any case :)

Reply:
I appreciate everyone weighing in on this.

I think that the choice to use the term "optional" to represent a union with `None` is unfortunate. My issue is that it has a meaning in lots of contexts, so when talking about typing through the prism of other domains, e.g., API endpoints/openapi schemas/function parameters, one has to be sure to convey what you mean by the term when using it in that context. E.g., making a parameter's type a union with `None`, doesn't make the parameter optional, the presence of a default value does. In an openapi context, optional could easily be conflated with a "not required" parameter. The old behavior of pydantic, where an explicit default for a `None` union wasn't required also blurred the lines around what "optional" means in both a typing, and parameter sense.

Given that we are split down the middle on this, and as I mentioned earlier in this thread - I recognize that it is a commonly accepted term in the python typing world, and as this is a library built around python's typing system, I'm OK with using it here.

So, I think status quo wins out, lets leave it as it is.